### PR TITLE
RegisterDevicePage: add UUIDv5 option

### PR DESCRIPTION
**Pull request by Pavinati**
Allow users to generate device IDs using UUIDv5

Signed-off-by: Mattia <mattia.pavinati@gmail.com>

### CODE REVIEW

File: src/react/RegisterDevicePage.js [hunk 1]
```diff
@@ -98,8 +166,7 @@ export default class RegisterDevicePage extends React.Component {
                 isInvalid={this.state.isInvalidDeviceId}
               />
               <Form.Control.Feedback type="invalid">
-                The device ID must be 22 characters long and only contain
-                letters, numbers and the following characters _ -
+                Device ID must be a unique 128 bit URL encoded base64 (without padding) string.
```
Comment (bettio) [inline src/react/RegisterDevicePage.js:169]: URL-encoded

File: src/react/RegisterDevicePage.js [hunk 2]
```diff
@@ -203,9 +278,64 @@ export default class RegisterDevicePage extends React.Component {
...
+
+  maybeGenerateDeviceId(namespace, customString) {
+    let newDeviceID = "";
+
+    try {
+      const newUUID = uuidv5(customString, namespace).replace(/-/g, "");
+      const bytes = newUUID.match(/.{2}/g).map(b => parseInt(b, 16));
+      newDeviceID = byteArrayToUrlSafeBase64(bytes);
+    } catch (e) {
+      // namespace is not a UUID
```
Comment (bettio) [inline src/react/RegisterDevicePage.js:325]: are you sure about this empty catch block?

Comment (Pavinati) [reply src/react/RegisterDevicePage.js:325]: i can make this more explicit, but yes, It is intended.

Comment (Pavinati) [reply src/react/RegisterDevicePage.js:325]: I'll rewrite this to make it clearer

**Pull request merged**